### PR TITLE
Added Third Party Flag To Events

### LINKED ISSUE #113: Non-TESC Organisation Event Labels
For non-TESC events, hosted on the platform, create labels on the application and user profile that indicate who is organising the event.

### PR DESCRIPTION
**Pull request by subhankar-panda**
Fixes #113

### CODE REVIEW

File: src/assets/js/pages/EventPage/index.js [hunk 1]
```diff
@@ -265,6 +266,16 @@ class EventPage extends React.Component {
                     </NavLink>
                   </NavItem>
                 ))}
+                {isThirdParty && 
+                  <NavItem className="ml-auto event-tab__alert my-auto">
+                    Third Party Event
+                    <span className="m-2">
+                      <FA name="question-circle" id={'ThirdPartyTooltip'}></FA>
+                      <UncontrolledTooltip placement="right" target={'ThirdPartyTooltip'}>
+                        This Event Is Not Run By TESC.
```
Comment (RedbackThomson) [inline src/assets/js/pages/EventPage/index.js]: Capitalisation is hard to read.
`This event is not run by TESC` is easier to read

File: src/assets/js/proptypes/index.js [hunk 2]
```diff
@@ -25,6 +25,8 @@ export const Event = {
   logo: PropTypes.shape({
     url: PropTypes.string.isRequired
   }).isRequired,
+  thirdPartyText: PropTypes.string.isRequired,
+  organisedBy: PropTypes.string.isRequired,
```
Comment (RedbackThomson) [inline src/assets/js/proptypes/index.js]: These two shouldn't be required. If they don't exist, then we assume TESC.

File: migrations/15-add_third_party_fields.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const organisedBy = 'TESC'
+module.exports.up = function (done) {
+    var events = this.db.collection('events');
+    events.updateMany({}, {$set: {thirdPartyText, organisedBy}}, done);
+};
+
+module.exports.down = function(done) {
+    var events = this.db.collection('events');
+    events.update({}, {$unset: {thirdPartyText: 1, organisedBy: 1}}, done);
+}
```
Comment (RedbackThomson) [inline migrations/15-add_third_party_fields.js]: If thirdPartyText is undefined then we can assume it is a TESC event. Therefore, it doesn't need to be in this migration.

File: src/assets/scss/components/_event-card.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
+.event-card {
+
+    &__header {
+        background: $tesc-blue;
+
+        font-weight: bold;
+        color: white;
+
+        &--third-party {
+            background: $medium-gray;
```
Comment (RedbackThomson) [inline src/assets/scss/components/_event-card.scss]: Gray seems a bit mean? Can we do another bold colour? Like $secondary-color ?

File: src/server/models/event.js [hunk 5]
```diff
@@ -51,6 +51,16 @@ var EventSchema = new Schema({
...
   },
+  thirdPartyText: {
+    type: String,
+    default: '',
+    trim: true
+  },
+  organisedBy: {
+    type: String,
+    default: 'TESC',
+    trim: true
```
Comment (RedbackThomson) [inline src/server/models/event.js]: `required: true`
If we have the default of TESC then we would expect everything to have a value.

File: src/assets/js/proptypes/index.js (hunk 2 above)
Comment (subhankar-panda) [reply src/assets/js/proptypes/index.js]: they'll always exist because of the mongoose defaults

File: src/assets/js/pages/EventPage/index.js [hunk 6]
```diff
@@ -209,13 +209,14 @@ class EventPage extends React.Component {
...
     let {event, statistics, alerts} = this.props;
     let {activeTab} = this.state;
-
     if (!event) {
       return (
         <Loading />
       );
     }
 
+    const isThirdParty = event.thirdPartyText.length > 0
```
Comment (RedbackThomson) [inline src/assets/js/pages/EventPage/index.js]: `const isThirdParty = !!event.thirdPartyText`

File: src/server/models/event.js [hunk 7]
```diff
@@ -51,6 +51,17 @@ var EventSchema = new Schema({
     trim: true,
     required: false
   },
+  thirdPartyText: {
+    type: String,
+    default: '',
```
Comment (RedbackThomson) [inline src/server/models/event.js]: No default, since undefined should be the default

**Pull request merged**